tests/storage/guestfs.go

```diff
@@ -119,6 +121,21 @@ var _ = SIGDescribe("[rfe_id:6364][[Serial]Guestfs", func() {
 
 	}
 
+	waitForPodToBeDeletedOrTerminated := func(podName string) {
```

Comment:
I think typically, for test speedup, instead of waiting for things to be deleted and cleaning up the environment between tests we prefer using unique names. What do you think about not waiting for the pod to be deleted or terminated at all and removing the AfterEach?

Reply:
This is for synchronizing with the function [runGuestfsOnPVC](https://github.com/kubevirt/kubevirt/blob/main/tests/storage/guestfs.go#L85). If the pod is cleaned-up before this function terminates, we might have a failure in the AfterEach complaining that the pod `libguestfs-tool-pvc-fail-to-run-twice` doesn't exist anymore for example.
Please note, I'm not attending that the pod is fully deleted but in `Terminating` state. This means that the channel has already been closed.
Basically, there is a race between each `defer f.closeChannel()` and the test clean-up